### ISSUE TITLE
Handle namespace conflicts for trackables

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13097]
### Describe your problem!

A user contacted us, as he got reports, that his "car trackable sticker" cannot be discovered using c:geo.

The problem here is, that his tracking code starts with `GK` and is thus interpreted by c:geo as the public code of a geokrety trackable (which coincidentally exists) and the geokret is opened instead.

### How to reproduce?

- Open search activity
- Enter a valid Groundspeak tracking code starting with GK into the trackable search

### Actual result after these steps?

The Geokret, whose public code is matching the Groundspeak trackable code is opened instead

### Expected result after these steps?

Short term solution(?):
This problem does also occur if the geokret service is NOT activated. In such cases there should not be any communication with geokret servers but rather only the active connectors should be used, leading in the correct trackable to be returned.

Generic solution:
Ideally requests should go out to all active trackable connectors in parallel and (in case of multiple matches) a selection be offered to the users, which trackable he wants to continue with.

Linked to #4822 (seems to describe the cause of this problem already).

### Reproducible

Yes

### c:geo Version

2022.06.06

**Comment by ztNFny** [comment]
The problem here is that all geocode-searches expect to return exactly 1 result - unlike string-searches. Adressing this would require rewriting the general search-logic to 1) get all results for a geocode, 2) if count == 1 show it, elseif count > 1 show a list of results (like for string searches)

Inside the trackable-search we first check whether the code provided is a geocode in any of the connectors (so anything starting with TB, GK, GC, OC, ...), only if that fails we check for a trackable-code.

The only realistic path I see for this (that has an okish effort-value ratio) is in src/cgeo/geocaching/connector/ConnectorFactory.java to initialize only enabled connectors.

**Comment by Lineflyer** [comment]
Over time more users contacted us with problems based on this issue:

There seem to be more and more tracking codes of Geocaching.com trackables existing out there which start with GK.
As GeoKrety is active in c:geo by default, once users try to look up the trackable via its tracking code the GeoKrety will show up instead and the user has no ability to load the GC trackable with c:geo....or at least I don`t know how (except he knows the public code TB....).

**Comment by moving-bits** [comment]
`ConnectorFactory` has a list of all connectors:

```
private static final Collection<TrackableConnector> TRACKABLE_CONNECTORS = Collections.unmodifiableCollection(Arrays.<TrackableConnector>asList(
    new GeokretyConnector(),
    TravelBugConnector.getInstance(), // travel bugs last, as their secret codes overlap with other connectors
    UNKNOWN_TRACKABLE_CONNECTOR // must be last
));
```

What about leaving out `new GeokretyConnector()` if it's not activated? Then no place should be calling `GeokretyConnector` anymore until it gets enabled.

Would probably need a restart of c:geo on changing the enabling state.

**Comment by moving-bits** [comment]
Implemented the partial workaround with PR #14056

**Comment by moving-bits** [comment]
Workaround is in the nightlies for quite some time, and IMHO it's not worth the effort to refactor TB handling to detect namespace conflicts (and present selection dialogs, if so) in case GK connector is activated, so I propose to close this issue.
